### PR TITLE
feat: msw setup

### PR DESCRIPTION
**Pull request by bhtibrewal**
### Dependency added : 
- @testing-library/jest-dom
- @vue/test-utils
- msw
- @vitest/coverage-v8

Made a folder for mock data, handlers 

image

### CODE REVIEW

File: adapters/goal.transformer.ts [hunk 1]
```diff
@@ -1,13 +1,15 @@
-import { type Goal } from '~/interfaces/Goal'
-import { PostGoal } from '~/interfaces/PostGoal'
+import { type Goal } from "~/interfaces/Goal";
+import { PostGoal } from "~/interfaces/PostGoal";
 
 export const transformGoalsFromApi = (goals: any): Goal[] => {
-  const transformedGoals = goals.map((goal: any) => transformGoalFromApi(goal))
-  return transformedGoals
-}
+  const transformedGoals = goals?.map((goal: any) =>
```
Comment (iamitprakash) [inline adapters/goal.transformer.ts:5]: what if the goals parameter is  undefined.

File: adapters/goal.ts [hunk 2]
```diff
@@ -1,49 +1,60 @@
...
   headers: {
-    'Content-Type': 'application/vnd.api+json'
-  }
-}
+    "Content-Type": "application/vnd.api+json",
+  },
+};
 
-export const fetchGoals = async (): Promise<Goal []> => {
+export const fetchGoals = async (): Promise<Goal[]> => {
```
Comment (iamitprakash) [inline adapters/goal.ts:13]: this function is not handling errors gracefully. If the axios request fails, the catch() block will throw an error, which will be propagated to the caller of the fetchGoals()

File: adapters/goal.ts [hunk 3]
```diff
@@ -1,49 +1,60 @@
...
+    .catch((error) => {
+      throw new Error(error);
+    });
 
-  return goals
-}
+  return goals;
+};
 
 export const addGoal = async (goal: PostGoal): Promise<Goal> => {
```
Comment (iamitprakash) [inline adapters/goal.ts:24]: here function is not checking the status code of the axios response. This means that if the axios request fails, the addGoal() function will return the goalResponse object, which may not be a Goal object.

File: adapters/goal.ts [hunk 4]
```diff
@@ -1,49 +1,60 @@
...
+    )
+    .then((res) => goalAdapter.transformGoalFromApi(res.data.data));
 
-  return goalResponse
-}
+  return goalResponse;
+};
 
-export const updateGoal = async (goalId: string, goal: PostGoal): Promise<Goal> => {
+export const updateGoal = async (
```
Comment (iamitprakash) [inline adapters/goal.ts:41]: here the function  is not checking if the goalId parameter is a valid ID. This means that if the goalId parameter is not a valid ID, the axios request will fail, and the updateGoal() function will not return a Goal object.

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: __mocks__/handlers/index.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { goalsHandlers } from './goals.handler'
+
+const handlers = [...goalsHandlers]
```
Comment (pallabez) [inline __mocks__/handlers/index.js:3]: Is there any reason on why we are doing this?

**Review by pallabez — APPROVED** [review]
LGTM

File: adapters/goal.ts (hunk 2 above)
Comment (pallabez) [reply adapters/goal.ts:13]: How should we handle this? Shouldn't caller of fetchGoals handle it?

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**